// ngx-highlight-js/bundles/ngx-highlight-js.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/forms')) :
    typeof define === 'function' && define.amd ? define('ngx-highlight-js', ['exports', '@angular/core', '@angular/common', '@angular/forms'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['ngx-highlight-js'] = {}, global.ng.core, global.ng.common, global.ng.forms));
}(this, (function (exports, core, common, forms) { 'use strict';

    var HIGHLIGHTJS_CONFIG = new core.InjectionToken('HighlightJs-Config');

    var HighlightJsDirective = /** @class */ (function () {
        function HighlightJsDirective(el, ngModel, doc, cog) {
            this.el = el;
            this.ngModel = ngModel;
            this.doc = doc;
            this.lang = 'html';
            this.mode = 'simple';
            Object.assign(this, cog);
        }
        HighlightJsDirective.prototype.escapeHTML = function (str) {
            return (str || '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;').replace(/"/g, '&quot;').replace(/'/g, '&apos;');
        };
        HighlightJsDirective.prototype.init = function () {
            this.destroy();
            var el = this.el.nativeElement;
            var code = this.code || '' + el.innerHTML.trim();
            this.codeEl = this.doc.createElement(this.mode === 'default' ? 'div' : 'pre');
            var isSimple = this.mode === 'simple';
            if (isSimple) {
                if (this.lang) {
                    this.codeEl.className = this.lang;
                }
                this.parentEl = el.parentNode;
                this.parentEl.insertBefore(this.codeEl, el.nextSibling);
            }
            else {
                this.parentEl = el;
                this.parentEl.innerHTML = "";
                this.parentEl.appendChild(this.codeEl);
            }
            this.codeEl.innerHTML = code;
            hljs.configure(Object.assign({}, this.options));
            if (isSimple) {
                hljs.highlightBlock(this.codeEl);
            }
            else {
                this.codeEl.querySelectorAll('pre code').forEach(function (block) {
                    hljs.highlightBlock(block);
                });
            }
        };
        HighlightJsDirective.prototype.destroy = function () {
            if (this.codeEl && this.parentEl) {
                this.parentEl.removeChild(this.codeEl);
                this.codeEl = undefined;
            }
        };
        HighlightJsDirective.prototype.ngAfterViewInit = function () {
            var _this = this;
            var _a;
            this.init();
            if (this.ngModel) {
                this.modelValue$ = (_a = this.ngModel.valueChanges) === null || _a === void 0 ? void 0 : _a.subscribe(function (res) {
                    _this.code = _this.escapeHTML(res);
                    _this.init();
                });
            }
            else {
                this.initMutation();
            }
        };
        HighlightJsDirective.prototype.ngOnDestroy = function () {
            this.destroy();
            this.destroyMutation();
            if (this.modelValue$) {
                this.modelValue$.unsubscribe();
            }
        };
        HighlightJsDirective.prototype.initMutation = function () {
            if (typeof MutationObserver === 'undefined') {
                return;
            }
            this.observer = new MutationObserver(this.init.bind(this));
            this.observer.observe(this.el.nativeElement, {
                characterData: true,
                childList: true,
                subtree: true,
            });
        };
        HighlightJsDirective.prototype.destroyMutation = function () {
            if (!this.observer) {
                return;
            }
            this.observer.disconnect();
        };
        return HighlightJsDirective;
    }());
    HighlightJsDirective.decorators = [
        { type: core.Directive, args: [{
                    selector: '[highlight-js]',
                    host: {
                        '[style.display]': "mode === 'simple' ? 'none' : null",
                    },
                    exportAs: 'highlightJs',
                },] }
    ];
    HighlightJsDirective.ctorParameters = function () { return [
        { type: core.ElementRef },
        { type: forms.NgModel, decorators: [{ type: core.Optional }] },
        { type: undefined, decorators: [{ type: core.Inject, args: [common.DOCUMENT,] }] },
        { type: undefined, decorators: [{ type: core.Optional }, { type: core.Inject, args: [HIGHLIGHTJS_CONFIG,] }] }
    ]; };
    HighlightJsDirective.propDecorators = {
        options: [{ type: core.Input }],
        lang: [{ type: core.Input }],
        code: [{ type: core.Input }],
        mode: [{ type: core.Input }]
    };

    var HighlightJsModule = /** @class */ (function () {
        function HighlightJsModule() {
        }
        return HighlightJsModule;
    }());
    HighlightJsModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [forms.FormsModule],
                    declarations: [HighlightJsDirective],
                    exports: [HighlightJsDirective],
                },] }
    ];

    /**
     * Generated bundle index. Do not edit.
     */

    exports.HIGHLIGHTJS_CONFIG = HIGHLIGHTJS_CONFIG;
    exports.HighlightJsDirective = HighlightJsDirective;
    exports.HighlightJsModule = HighlightJsModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
